### PR TITLE
[utils] generic vendor server

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR include two commits:
1. reverts https://github.com/openthread/ot-br-posix/pull/1967 as it breaks community use cases
2. Make VendorServer more generic to be used by multiple vendors. See https://github.com/openthread/ot-br-posix/pull/1967#issuecomment-1702407419

------

Prefer merging this PR as separate commits

### CODE REVIEW

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Comment by wgtdkp** [comment]
@jwhui prefer merging this PR as separate commits

File: src/agent/vendor.hpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ */
+class VendorServer
+{
+public:
+    virtual ~VendorServer(void) = default;
+
+    /**
+     * Creates a new instance of VendorServer.
+     *
+     * Custom vendor servers should implement this method to return an object of the derived class.
```
Comment (jwhui) [inline src/agent/vendor.hpp:55]: Document arguments and return value?

Comment (wgtdkp) [reply src/agent/vendor.hpp:55]: done

**Review by jwhui — APPROVED** [review]

**Pull request merged**